### ISSUE TITLE
8.4.4. New - Screen Name/Title: Browse templates - Copy - Not Programmatically Marked as Such

### ISSUE THREAD

**Issue by jzbahrai** [cds-snc/notification-planning#2465]
Search Field is Optional, but Not Programmatically Marked as Such
Missing Error Handling or Empty State for Search
Components: Copy an existing template - Search by name (Optional)
Observation:
1. The form visually indicates the search input is optional (with “(Optional)”), but this is not communicated to screen readers.
2. There is no visible or programmatic indication of how to handle no search results or invalid searches.

(Attachment: 
8.4.4. Search Field is Optional, but Not Programmatically Marked as Such.png)

Recommendation: 
Add " (optional)" to the label programmatically for screen reader users:
Example:
<label class="form-label" for="search">
  Search by name <span class="sr-only">(optional)</span>
</label>
2. When there are no matches, display a clear message like:
Example:
<p role="alert" id="no-results">No templates match your search.</p>
And consider associating it with aria-live="polite" region to notify screen reader users.

**Comment by whabanks** [comment]
We nest a span in the label for the search bar. When navigating element by element, the `search by name` label will not include the optional text. However selecting the textbox does in fact read the optional text.
```
<label class="form-label" for="search">
       Search by name
      <span id="search-optional" class="form-hint inline-block">
        (Optional)
      </span>
</label>
```
